File: frontend/src/types/cultura.ts
/** TypeScript interfaces for Cultura entity matching Pydantic schemas. */

export interface CulturaResponse {
  id: number;
  nome: string;
  descricao: string | null;
  ativo: boolean;
  created_at: string;
  updated_at: string;
}

export interface CulturaCreate {
  nome: string;
  descricao?: string | null;
}

export interface CulturaUpdate {
  nome?: string | null;
  descricao?: string | null;
}

File: app/schemas/cultura.py
"""Pydantic schemas for Cultura entity."""

from datetime import datetime

from pydantic import BaseModel, Field


class CulturaBase(BaseModel):
    nome: str = Field(..., max_length=120)
    descricao: str | None = None


class CulturaCreate(CulturaBase):
    pass


class CulturaUpdate(BaseModel):
    nome: str | None = Field(None, max_length=120)
    descricao: str | None = None


class CulturaResponse(CulturaBase):
    id: int
    ativo: bool
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
